index.js

```diff
@@ -1 +1,121 @@
 // implement your API here
+
+//this is how you import in express
+const express = require("express");
+const db = require("./data/db");
+
+const server = express();
+server.use(express.json());
+
+//http is protocol that connects client and server
+//send is a node function while get is http
+server.get("/", (req, res) => {
+  res.send("Hello World");
+});
+
+//GET 
+
+server.get("/api/users", (req, res) => {
+  db.find()
+    .then(users => {
+      if (users) {
+        res.status(200).json(users);
+      } else {
+        res
+          .status(404)
+          .json({ message: "The users information could not be retrieved." });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: "The user information could not be retrieved." });
+    });
+});
+
+//GET ID
+
+server.get("/api/users/:id", (req, res) => {
+  const id = req.param.id;
+  db.findById(id)
+    .then(user => {
+      if (user) {
+        res.status(200).json(user);
+      } else {
+        res
+          .status(404)
+          .json({ message: "The user with the specified ID does not exist." });
+      }
+    })
+    .catch(err => {
+      res
+        .status(500)
+        .json({ error: "The user information could not be retrieved." });
+    });
+});
+
+//ADD
+
+server.post("/api/users", (req, res) => {
+  const user = req.body;
+
+  db.insert(user)
+    .then(user => {
+      if (user.name && user.name) {
```

Comment:
If you look at that insert function, you can see what it will be returning back to you in the promise:

function insert(user) {
  return db('users')
    .insert(user)
    .then(ids => ({ id: ids[0] }));
}

So you're not actually getting back the whole user. You're just getting back an object that has the id of the newly created user. On the line where you wrote:

  .then(user => {

That user is actually a fubar banana word that represents the thing being returned from the db.insert() function. It might help to slip a console.log in there to give you an idea of what you're getting back.